numpy/core/tests/test_multiarray.py

```diff
@@ -4333,7 +4333,41 @@ def test_object_argmax_with_NULLs(self):
         assert_equal(a.argmax(), 3)
         a[1] = 30
         assert_equal(a.argmax(), 1)
+    
+    def test_np_argmax_keepdims(self):
 
+        sizes = [(3,), (3, 2), (2, 3),
+                 (3, 3), (2, 3, 4), (4, 3, 2),
+                 (1, 2, 3, 4), (2, 3, 4, 1),
+                 (3, 4, 1, 2), (4, 1, 2, 3)]
+        for size in sizes:
+            arr = np.random.normal(size=size)
+            for axis in range(-len(size), len(size)):
```

Comment:
This could also just be parametrized, and I think you may be able to just include `None` into that.

Reply:
I don't think that would be possible. This would lead to pairing of wrong list of axes and size since parametrization takes a cross product of all the arguments. For example `(3, 2)` can get mapped with `[-4, -3, -2, -1, 0, 1, 2, 3]`.

Reply:
I have parametrized this test. Let me know if the logic works. Thanks.